src/sqs/backfill-error-handler.test.ts

```diff
@@ -116,4 +118,30 @@ describe("backfillErrorHandler", () => {
 		expect(sendMessageMock.mock.calls[0][1]).toEqual(0);
 		expect((await RepoSyncState.findByPk(repoSyncState!.id)).commitStatus).toEqual("failed");
 	});
+
+	it("marks task as failed and reschedules message on permission error", async () => {
+		const { subscription, repoSyncState } = await new DatabaseStateCreator().withActiveRepoSyncState().create();
+
+		const context = createContext(5, true);
+		context.payload = {
+			jiraHost,
+			installationId: subscription.gitHubInstallationId
+		};
+
+		const task = _.cloneDeep(TASK);
+		task.repositoryId = repoSyncState?.repoId || -1;
+
+		const sendMessageMock = jest.fn();
+		const result = await backfillErrorHandler(sendMessageMock)(new TaskError(task, new InvalidPermissionsError({ } as unknown as AxiosError)), context);
+
+		expect(result).toEqual({
+			isFailure: false
```

Comment:
should it expect `isRetryable: true` as well?

Reply:
It doesn't matter, because ifFailure means that the error is not considered as a "error" and the message will be deleted.